admin/src/components/Federation/CommunityVisualizeItem.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+  <div class="community-visualize-item">
+    <b-row @click="toggleDetails">
+      <b-col cols="1"><b-icon :icon="icon" :variant="variant" class="mr-4"></b-icon></b-col>
+      <b-col>
+        <div>
+          <a :href="item.url" target="_blank">{{ item.url }}</a>
+        </div>
+        <small>{{ `${item.publicKey.substring(0, 26)}…` }}</small>
+      </b-col>
+      <b-col v-b-tooltip="item.description">{{ item.name }}</b-col>
+      <b-col cols="2">{{ lastAnnouncedAt }}</b-col>
+      <b-col cols="2">{{ createdAt }}</b-col>
+    </b-row>
```

Comment:
please see my comment in ...graphql/allCommunities.js

The ui-component to render the values for one _community_ have to be designed to treat the database constellation of existing `federatedCommunities`-entries but missing `communities`-entry and several `federatedCommunities`-entries for the same `publicKey`

Reply:
ChatGPT 3.5:

In einem gut strukturierten System sollten die verschiedenen Komponenten jeweils für ihre spezifischen Aufgabenbereiche verantwortlich sein:

#### Backend (Datenbankkonstellation):

- Das Backend, einschließlich der Datenbank und der entsprechenden Logik, sollte sich um die Verwaltung der Datenbankkonstellation kümmern. Dazu gehört die Speicherung und Verarbeitung von Daten gemäß den Anforderungen und Geschäftslogik der Anwendung.
- Das Backend ist dafür verantwortlich sicherzustellen, dass die Daten konsistent sind und in einer für das Frontend verwertbaren Form vorliegen.

#### GraphQL (Datenformat und Abfragen):

- GraphQL ist eine Abfragesprache und Ausführungsumgebung für APIs, die es dem Frontend ermöglicht, genau die Daten abzurufen, die es benötigt.
- GraphQL dient als Zwischenschicht zwischen dem Frontend und dem Backend und ermöglicht es dem Frontend, flexible Abfragen zu stellen, um genau die Daten zu erhalten, die es benötigt.
- GraphQL kann die Datenbankkonstellation im Backend verbergen und sicherstellen, dass das Frontend nur die benötigten Daten im erwarteten Format erhält.

#### Frontend (Darstellung und Benutzerinteraktion):

- Das Frontend ist dafür verantwortlich, die Daten, die es vom Backend über GraphQL erhält, auf angemessene Weise darzustellen und dem Benutzer eine benutzerfreundliche Benutzeroberfläche zu bieten.
- Das Frontend sollte sich nicht um die internen Details der Datenbankkonstellation oder der GraphQL-Abfragen kümmern müssen, sondern einfach die Daten im erwarteten Format erhalten und anzeigen können.

Indem jede Komponente ihre spezifischen Aufgabenbereiche hat und sich auf diese konzentriert, kann eine klare Trennung der Verantwortlichkeiten erreicht werden, was zu einer einfacheren Entwicklung, Wartung und Skalierbarkeit des Systems führt.

Reply:
> The ui-component to render the values for one community have to be designed to treat the database constellation of existing federatedCommunities-entries but missing communities-entry and several federatedCommunities-entries for the same publicKey

you can try it, but I found it really hard to archive this with vue2, it was much easier todo in backend.